src/dash-table/reconcile/date.ts

```diff
@@ -0,0 +1,74 @@
+import { IDatetimeColumn } from 'dash-table/components/Table/props';
+import { reconcileNull } from './null';
+import { IReconciliation } from '.';
+
+// pattern and convertToMs pulled from plotly.js
+// (simplified - no international calendars for now)
+// https://github.com/plotly/plotly.js/blob/master/src/lib/dates.js
+// Note we allow timezone info but ignore it - at least for now.
+const DATETIME_REGEXP = /^\s*(-?\d\d\d\d|\d\d)(-(\d?\d)(-(\d?\d)([ Tt]([01]?\d|2[0-3])(:([0-5]\d)(:([0-5]\d(\.\d+)?))?(Z|z|[+\-]\d\d:?\d\d)?)?)?)?)?\s*$/m;
```

Comment:
Any reason for not using the terser \d{4}, \d{2}, \d{1,2}?

Same question for the 'i' flag to make it case insensitive (T|t, Z|z)

Reply:
No reason, just copied over from plotly.js. I could tighten up the `\d` usage here, though I might prefer to keep `[ Tt]` and `(Z|z|...` over `/i`, seems more explicit since letters are a very small fraction of the overall pattern.

Reply:
Kk, I'm fine with it as is in any case.

Reply:
`/k/i`